compiler/rustc_middle/src/mir/interpret/value.rs

```diff
@@ -266,6 +266,16 @@ impl<Prov> Scalar<Prov> {
             .unwrap_or_else(|| bug!("Signed value {:#x} does not fit in {} bits", i, size.bits()))
     }
 
+    #[inline]
+    pub fn from_i8(i: i8) -> Self {
+        Self::from_int(i, Size::from_bits(8))
+    }
+
+    #[inline]
+    pub fn from_i16(i: i16) -> Self {
+        Self::from_int(i, Size::from_bits(16))
+    }
```

Comment:
If you're adding these, can you also add the unsigned variants while you are at it? Let's keep our method set somewhat internally consistent.

Reply:
Unsigned variants were already there.